include/eve/arch/riscv/as_register.hpp

```diff
@@ -0,0 +1,187 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/traits/as_integer.hpp>
+
+#include <type_traits>
+#if defined(EVE_HW_RISCV_SVE)
+namespace eve
+{
+template<typename T> struct logical;
+}
+
+namespace eve
+{
+template<typename Type, typename Size, rvv_abi ABI> struct as_register<Type, Size, ABI>
+{
+  static constexpr auto lmul = riscv_rvv_dyn_::getLMUL<Type>(Size::value);
+
+  static constexpr bool is_fp_v     = std::is_floating_point_v<Type>;
+  static constexpr bool is_signed_v = std::is_signed_v<Type>;
+
+#  ifdef EVE_RISCV_REG_CHOOSE
```

Comment:
heavy macro usage seems unjustified on a surface

Reply:
I am trying to create a group of functions to make the best decision for vector type depending on requested cardinal/type and sew(single element width). For example, let's suppose `vlen == 128`:
1) you need to work with 16 int8 elements - you can use vector type `vint8m1_t`
2) you need to work with 128 uint8 elements - you can use vector type `vuint8m8_t`
3) you need to work with 8 float elements - you can use vector type `vfloat32m4`